tests/integration/config/plugins/integration-tests/data/tests/tests_conditional_choice.txt

```diff
@@ -0,0 +1,188 @@
+				to display
+					not "%TEST%: CONDITIONAL CHOICE"
+			`"Does this work? Do you not see the first sentence?"`
+			choice
+				`	"No."`
+					to display
+						has "%TEST%: CONDITIONAL CHOICE"
+					accept
+				`	"Yes."`
+					decline
+	on accept
+		set "succeed test"
+
+mission "%TEST%: CONDITIONAL CHOICE GOTO"
+	priority
+	source "Ruin"
+	to offer
+		has "%TEST%: CONDITIONAL CHOICE GOTO"
+	on offer
+		conversation
+			`"Can you use the hidden path?"`
+			choice
+				`	"Yes."`
+					to display
+						not "%TEST%: CONDITIONAL CHOICE GOTO"
+				`	"No."`
+					to display
+						has "%TEST%: CONDITIONAL CHOICE GOTO"
+					goto canuse
+			# just in case the goto is fully ignored
+			`	"You should not be here!"`
+				decline
+			label canuse
+			`	"Some paths are meant to stay hidden."`
+				accept
+	on accept
+		set "succeed test"
+
+
+mission "%TEST%: NAME IN CONVERSATION"
+	priority
+	source "Ruin"
+	to offer
+		has "%TEST%: NAME IN CONVERSATION"
+	on offer
+		conversation
+			name
+			`	So you are <first> <last>.`
+
+
+test "Trigger Spaceport Mission"
+	status partial
+	description "Launch, land, and go to the spaceport to trigger any potential missions."
+	sequence
+		call "Depart"
+		call "Land On Ruin helper"
+		input
+			key p
+		# press accept (or decline, depending on if it is shown) and then skip the dialog once
+		input
+			key "Return"
+		input
+			key "Return"
+
+
+test "Conditional Test"
+	status active
+	description "Test if a conversation option shows up when the given condition is set."
+	sequence
+		inject "conditional choice"
+		call "Load First Savegame"
+		apply
+			"%TEST%: CONDITIONAL CHOICE" = 1
+		call "Trigger Spaceport Mission"
+		call "Depart"
+		assert
+			"succeed test" == 1
+
+
+test "Conditional Test Goto"
+	status active
+	description "Test if a first conversation option does not show up but the second one does, when the given condition is set. Also makes sure goto still works as expected."
+	sequence
+		inject "conditional choice"
+		call "Load First Savegame"
+		apply
+			"%TEST%: CONDITIONAL CHOICE GOTO" = 1
+		call "Trigger Spaceport Mission"
+		call "Depart"
+		assert
+			"succeed test" == 1
+
+
+test "Enter Name in Conversation"
+	status active
+	description "Test if a first conversation option does not show up but the second one does, when the given condition is set. Also makes sure goto still works as expected."
+	sequence
+		inject "conditional choice"
+		call "Load First Savegame"
+		apply
+			"%TEST%: NAME IN CONVERSATION" = 1
+		call "Trigger Spaceport Mission"
+		input
+			key F
+		input
+			key "Return"
+		input
+			key L
+		input
+			key "Return"
+		input
+			key "Return"
+		call "Depart"
+		# use a test-data temporary variable for debugging in case of errors
+		apply
+			"test: %TEST%: NAME IN CONVERSATION: offered" = "%TEST%: NAME IN CONVERSATION: offered"
+			"test: <first>" = "<first>"
+			"test: <last>" = "<last>"
+		assert
+			"%TEST%: NAME IN CONVERSATION: offered" == 1
+			"<first>" == "F"
```

Comment:
Variable `"<first>"` is not set, so it will have a default value of 0. And conditions don't support strings, so I expect the `"F"` to be converted to a value of 0 as well. (The comparison returns true then, but doesn't really test anything.)
Same for the `"<last>"` variable; those are available as text-substitutions, but not as conditions.

Reply:
Btw, Derpy recently added an option to check the players name, then conditions should be something like `"first name: F" == 1` and similar for last-name.

Reply:
Well the goal wasnt to test that command I just thought lets go ahead and do it. 
I do have another PR that does just that though, so even if this one test does not work we'll figure it out in that other PR.

The main point of it was that you get a `name` thing happening and not a blank screen, I just thought may as well.

Now, in order to check for that still, we can just do it in an on accept branch, somehow?

Reply:
> Btw, Derpy recently added an option to check the players name, then conditions should be something like `"first name: F" == 1` and similar for last-name.

Yeah we should totally use that.
I know "name: F L" would work in this case, not sure about first name but it should work as well if it exists.

Reply:
[It exists](https://github.com/endless-sky/endless-sky/blob/d925975124dcf3ae67760b161b5e4459056ace5d/source/PlayerInfo.cpp#L2759)